### ISSUE TITLE
maxPropArrayLength not working ( globally, per story, per stories)

### ISSUE THREAD

**Issue by kushalmahajan** [storybookjs/storybook#8777]
Version -  5.2.3 
I have a Button component

```
export const Button: React.FC<ButtonProps> = ({
	btnType = "outlined",
	btnStyle = "default",
	icon,
	loading = false,
	onClick = function() {},
	href = "#",
	target = "blank",
	htmlType = "button",
	children = <></>,
	...rest
}: ButtonProps) => {
    ...
    return (
        <ButtonDefault
            onClick={handleClick}
            btnType={btnType}
            btnStyle={btnStyle}
            type={htmlType}
            {...rest}
        >
            {icon ? getIcon(btnStyle, btnType) : null}
            {children}
        </ButtonDefault>
    )
    
```

`ButtonDefault` is a styled component of the signature 

```
export const ButtonDefault = styled.button<ButtonProps>`

```

In *config.js*

```
addDecorator(withInfo({
  infoBody: {
    padding: '0px 15px',
  },
  inline: true
}));
```

My story is 

```
const stories = storiesOf("Buttons", module)
.add(
    "outlined",
    () => (
        <StoryWrapper>
            <Row>
                <Col sm={3} md={3} lg={3}>
                    <Button size="lg">Default outlined</Button>
                </Col>
                <Col sm={3} md={3} lg={3}>
                    <Button disabled size="lg">
                        Disabled outlined
                </Button>
                </Col>
                <Col sm={3} md={3} lg={3}>
                    <Button btnStyle="primary" size="lg">
                        Primary outlined
                </Button>
                </Col>
                <Col sm={3} md={3} lg={3}>
                    <Button btnStyle="danger" size="lg">
                        Danger outlined
                </Button>
                </Col>
            </Row>
        </StoryWrapper>
    ),
    {
        maxPropArrayLength: 10
    }
)
```

First, despite this setup, it renders StoryWrapper prop table is displayed.
Second, both the StoryWrapper and Button component includes a hell of a lot of props. I would like to only display props defined in an interface for Button

Seems like local overrides are not working.

My storybook is hosted here - https://turtlemint-ui.netlify.com/

**Comment by kushalmahajan** [comment]
In fact, `withInfo` is not working at all.

*config.js*

```
import { addDecorator } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { addParameters, configure } from '@storybook/react';
import requireContext from 'require-context.macro';

addDecorator(withInfo({
  infoBody: {
    padding: '0px 115px',
  },
  inline: true,
  maxPropArrayLength: 15
}));

const req = requireContext('../src/', true, /\.stories\.tsx$/);

function loadStories() {
    req.keys().forEach(req);
}

configure(loadStories, module);
```

**Comment by shilman** [comment]
`Addon-info` is being superceded by `addon-docs`, which fixes a bunch of bugs and is easier to maintain. Please give it a try! https://medium.com/storybookjs/storybook-docspage-e185bc3622bf

**Comment by kushalmahajan** [comment]
Cool, I'll take up addon-docs in v2 of my development maybe. 

Coming to the main issue. I'm playing around and I did the following. 

1. Updated the storybook to 5.2.6
2. Corrected my `config.js` shape to 
```
addDecorator(withInfo({
  	inline: true,
    styles: {
      infoStory: {
        padding: 30
      }
    },
    maxPropArrayLength: 15
}));
```
3. Corrected the placement of the params of info which are supposed to be inside `info` object itself.

```
info: {
    propTables: [Button],
    maxPropArrayLength: 10,
    excludedPropTypes: ["aria-activedescendant", "aria-atomic", "onPointerUp"]
}
```
Apologies! 

Now,  all this said `maxPropArrayLength` is still not working. Checked globally and in combined story form i.e.

```
const stories = storiesOf("Buttons", module).addParameters({
	info: {
		TableComponent,
		propTables: [Button],
		maxPropArrayLength: 10,
		excludedPropTypes: ["aria-activedescendant", "aria-atomic", "onPointerUp"]
	}
});
```

My understanding from https://github.com/storybookjs/storybook/tree/next/addons/info page is that `maxPropArrayLength` Displays the first 10 items in the default prop array
which is not happening. The table displays all the props from various addon ( a11y et cetera et cetera);

Also, there is no way to control the entries.  `excludedPropTypes` is the property for that I know.

```
excludedPropTypes: ["aria-activedescendant", "aria-atomic", "onPointerUp", ... , ....]
```
But the list can go on and on with the number of props rendered by default. Trust me, I don't wanna count it. It's that long.

The alternative is to provide my own `TableComponent` for now, inside `info` object which can control the no of entries for display.

Raw thoughts - `props.slice(0, 10)` inside `TableComponent` does the work.

Now, since the original question title is not a bug. I'll change the title for `maxPropArrayLength`. Ofcourse if my understanding is correct of its usage from the English mentioned in docs.
